src/core/channels/presence_channel.ts

```diff
@@ -52,10 +52,17 @@ export default class PresenceChannel extends PrivateChannel {
       this.handleInternalMessage(message)
     } else {
       var data = message.data;
-      if (message.user_id) {
+
+      if (event.indexOf("client-") === 0) {
+
+        // client events on presence channels include a user_id field in the top
+        // level of the message object. This user_id parameter should be
+        // included in a metadata object included as the second argument to
+        // bound callbacks
         var metadata = {};
         metadata['user_id'] = message.user_id;
```

Comment:
If we make this configurable via a server-side switch then this field could be `null`.

Reply:
This is no longer relevant 👍